source/transfer_list.rst

```diff
@@ -84,12 +86,12 @@ version field determines the contents of the handoff start header.
    * - size
      - 0x4
      - 0x8
-     - The number of bytes occupied by the TL. This field accounts for the size of the TL header plus the size of all the entries contained in the TL. This field must be updated when any entry is added to the TL.
+     - The number of bytes occupied by the TL. This field accounts for the size of the TL header plus the size of all the entries contained in the TL. It must be a multiple of 16 (i.e. it counts the inter-TE padding after the end of the last TE). This field must be updated when any entry is added to the TL.
 
    * - max_size
      - 0x4
      - 0xc
-     - The maximum number of bytes that the TL can occupy. Any entry producer must check if there is sufficient space before adding an entry to the list. Firmware can resize and/or relocated the TL and update this field accordingly, provided that the TL requirements are respected.
+     - The maximum number of bytes that the TL can occupy. Any entry producer must check if there is sufficient space before adding an entry to the list. Firmware can resize and/or relocated the TL and update this field accordingly, provided that the TL requirements are respected. This field must be a multiple of 16. Bytes between tl_base_pa + size and tl_base_pa + max_size (i.e. the unallocated extra space that more TEs can be added to) do not need to be initialized to any specific value.
 
```

Comment:
"It also adds a new requirement that the padding between TEs must be set
to zero (since this padding is included in the checksum calculation,
it's probably a good idea to clearly define it). "

I don't see this in the text, but I don't think it is necessary either. It is a pain to have to clear data that is not part of the structure being used.

Reply:
I'm not certain that we need mandating padding areas to be zero'd
The checksum computation takes into account the the data in the padding sections: 
https://github.com/FirmwareHandoff/firmware_handoff/blame/b5853dc18c3df287b542e8a95eecad465f3d3c0d/source/transfer_list.rst#L69
"The checksum is set to a value such that the sum over every byte in the {tl_base_pa + hdr_size, …, tl_base_pa + size - hdr_size -1}"

Reply:
Okay, I just felt that calculating a checksum over uninitialized bytes feels quite weird, but if we prefer that over extra initialization I guess there's nothing technically problematic about it. Skipping the padding bytes in the checksum would make checksum verification more complicated because you'd have to basically parse the whole list with hdr_size and data_size for each entry (rather than just treating the whole list as a big blob of bytes), so I'm not sure that's the best option.

Reply:
OK, I'm happy include uninitialized bytes in the checksum then. I agree it's a bit weird but the point of the checksum is to ensure a TL is passed from A to B correctly, not to verify its contents.

Reply:
Updated in latest push.